dingo/pipe/sampling.py

```diff
@@ -136,6 +137,8 @@ def _load_sampler(self):
 
         self.dingo_sampler.context = self.context
         self.dingo_sampler.event_metadata = self.event_metadata
+        # Update frequency ranges of sampler
+        self.dingo_sampler.sampling_updates = self.sampling_updates
```

Comment:
So this code recalls `._initialize_transforms` which does add the MaskingFrequency transform. However, it does not add a MaskingFrequency transform to the init_sampler.

Reply:
I think one can do `init_sampler.sampling_updates = self.sampling_updates` before initializing the GWSamplerGNPE to fix this.

Reply:
I made a few more changes. One thing I'm noticing is that here: https://github.com/dingo-gw/dingo/blob/main/dingo/core/samplers.py#L429 the data is being masked AND cast to a tensor of size (batch_size, num_det, 2, num_frequency_bins). Then since one is passing a tensor to the main network trasforms, you get an error here: https://github.com/dingo-gw/dingo/blob/strain_cropping/dingo/gw/transforms/waveform_transforms.py#L430 (since it's a tensor which doesn't have keys). 

I think the expected behaviour should be to not add the sampling updates to the main sampler if the init_sampler has strain_cropping and GNPE is used. I've implemented this locally and it seems to work, if you agree I can push it.